flink-queryable-state/flink-queryable-state-client-java/src/main/java/org/apache/flink/queryablestate/network/Client.java

```diff
@@ -483,27 +511,31 @@ private boolean close(Throwable cause) {
 		@Override
 		public void onRequestResult(long requestId, RESP response) {
 			TimestampedCompletableFuture pending = pendingRequests.remove(requestId);
-			if (pending != null && pending.complete(response)) {
+			if (pending != null && !pending.isDone()) {
 				long durationMillis = (System.nanoTime() - pending.getTimestamp()) / 1_000_000L;
 				stats.reportSuccessfulRequest(durationMillis);
```

Comment:
Shouldn't we only increase the `stats.reportSuccessfulRequest` if we could complete `pending`?

Reply:
This was creating test instabilities in the `AbstractServerTest.testPortRangeSuccess()`. This is why I decided to complete the future after increasing the counter. Given that we check if the `pending.isDone()`, we do not leave much room for false increases.